paasta_tools/kubernetes_tools.py

```diff
@@ -570,18 +570,25 @@ def namespace_external_metric_name(self, metric_name: str) -> str:
     def get_autoscaling_metric_spec(
         self, name: str, cluster: str, namespace: str = "paasta"
     ) -> Optional[V2beta1HorizontalPodAutoscaler]:
+        # Returns None if an HPA should not be attached based on the config,
+        # or the config is invalid.
+
+        if self.get_desired_state() == "stop":
+            return None
+
+        autoscaling_params = self.get_autoscaling_params()
+        if autoscaling_params["decision_policy"] == "bespoke":
+            return None
+
         # use new autoscaling configuration if it exists.
         if "horizontal_autoscaling" in self.config_dict:
             return self.get_hpa_metric_spec(name, cluster, namespace)
+
```

Comment:
Nit: can we change the order of these two blocks so that we check the decision policy after we check for the `horizontal_autoscaling` key?  The `decision_policy` only applies to the "legacy" `autoscaling` section, and this is meant to defer to the `horizontal_autoscaling` section if it exists.  So it doesn't really make sense to check a key in the legacy `autoscaling` section and then defer to `horizontal_autoscaling`.

Reply:
Sure, done.